endpoint/healthz/endpoint.go

```diff
@@ -0,0 +1,115 @@
+package healthz
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	kitendpoint "github.com/go-kit/kit/endpoint"
+	kithttp "github.com/go-kit/kit/transport/http"
+
+	"github.com/giantswarm/microendpoint/service/healthz"
+)
+
+const (
+	// Method is the HTTP method this endpoint is register for.
+	Method = "GET"
+	// Name identifies the endpoint. It is aligned to the package path.
+	Name = "healthz"
+	// Path is the HTTP request path this endpoint is registered for.
+	Path = "/healthz"
+)
+
+// Config represents the configured used to create a healthz endpoint.
+type Config struct {
+	// Dependencies.
+	Logger   micrologger.Logger
+	Services []healthz.Service
+}
+
+// DefaultConfig provides a default configuration to create a new healthz
+// endpoint by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		Logger:   nil,
+		Services: nil,
+	}
+}
+
+// New creates a new configured healthz endpoint.
+func New(config Config) (*Endpoint, error) {
+	// Dependencies.
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+	if len(config.Services) == 0 {
+		return nil, microerror.Maskf(invalidConfigError, "config.Services must not be empty")
+	}
+
+	newEndpoint := &Endpoint{
+		logger:   config.Logger,
+		services: config.Services,
+	}
+
+	return newEndpoint, nil
+}
+
+type Endpoint struct {
+	// Dependencies.
+	logger   micrologger.Logger
+	services []healthz.Service
+}
+
+func (e *Endpoint) Decoder() kithttp.DecodeRequestFunc {
+	return func(ctx context.Context, r *http.Request) (interface{}, error) {
+		return nil, nil
+	}
+}
+
+func (e *Endpoint) Encoder() kithttp.EncodeResponseFunc {
+	return func(ctx context.Context, w http.ResponseWriter, response interface{}) error {
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
+
+		return json.NewEncoder(w).Encode(response)
+	}
+}
+
+func (e *Endpoint) Endpoint() kitendpoint.Endpoint {
```

Comment:
So this is the general endpoint every of our microservices can import and use to execute health checks. We can provide common health checks like the `k8s` one I implemented with this PR, or users also hook up their own ones. The health checks can simply be composed. I like this.